wrangler-core/src/main/java/io/cdap/wrangler/utils/ArithmeticOperations.java

```diff
@@ -422,18 +422,16 @@ public static Number average(Number... nums) throws DirectiveExecutionException
       if (num instanceof Byte) {
         throw new DirectiveExecutionException("Input cannot be of type 'Byte'.");
       }
-      // Compute average (no BigDecimal case)
+      // Switch to BigDecimal computation if necessary
       if (noBigDecimal && num instanceof BigDecimal) {
         if (t > 1) {
           avg = BigDecimal.valueOf((double) avg);
         } else {
           avg = BigDecimal.ZERO;
         }
         noBigDecimal = false;
-      } else if (noBigDecimal) {
-        avg = (double) avg + (num.doubleValue() - (double) avg) / t;
       }
-      // Compute average (BigDecimal case)
+      // Compute average
       if (!noBigDecimal) {
```

Comment:
nitpick: not...no bigDecimal? Can we change this variable name to `isBigDecimal` or `containsBigDecimal` or something like this?

Reply:
Sure, sounds good. I'll be changing the variable name to containsBigDecimal.

Reply:
Done. Thank you!